openedx/core/djangoapps/appsembler/api/v2/views.py

```diff
@@ -0,0 +1,246 @@
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.core.djangoapps.user_authn.views.register import create_account_with_params
+from openedx.core.djangoapps.user_authn.views.password_reset import PasswordResetFormNoActive
+from student.helpers import AccountValidationError
+
+from student.models import CourseEnrollment
+
+from openedx.core.djangoapps.appsembler.api.v1.mixins import TahoeAuthMixin
+
+from lms.djangoapps.instructor.enrollment import (
+    enroll_email,
+    get_email_params,
+    unenroll_email,
+)
+
+from openedx.core.djangoapps.appsembler.api.helpers import as_course_key
+from openedx.core.djangoapps.appsembler.api.v1.api import (
+    enroll_learners_in_course,
+    unenroll_learners_in_course,
+)
+from openedx.core.djangoapps.appsembler.api.v2.api import (
+    account_exists
+)
+from openedx.core.djangoapps.appsembler.api.v1.filters import (
+    CourseEnrollmentFilter,
+    CourseOverviewFilter,
+    UserIndexFilter,
+)
+from openedx.core.djangoapps.appsembler.api.v1.pagination import (
+    TahoeLimitOffsetPagination
+)
+from openedx.core.djangoapps.appsembler.api.v1.serializers import (
+    CourseOverviewSerializer,
+    BulkEnrollmentSerializer,
+    UserIndexSerializer,
+)
+from openedx.core.djangoapps.appsembler.api.v1.waffle import FIX_ENROLLMENT_RESULTS_BUG
+
+# TODO: Just move into v1 directory
+from openedx.core.djangoapps.appsembler.api.permissions import (
+    TahoeAPIUserThrottle
+)
+from openedx.core.djangoapps.appsembler.api.sites import (
+    get_courses_for_site,
+    get_site_for_course,
+    get_enrollments_for_site,
+    course_belongs_to_site,
+    get_users_for_site,
+)
+
+
+log = logging.getLogger(__name__)
+
+#
+# Helper functions
+#
+
+
+def create_password():
+    """
+    Copied from appsembler_api `CreateUserAccountWithoutPasswordView`
+    """
+    return ''.join(
+        random.choice(
+            string.ascii_uppercase + string.ascii_lowercase + string.digits)
+        for _ in range(32))
+
+
+@beeline.traced(name="apis.v1.views.send_password_reset_email")
+def send_password_reset_email(request):
+    """Copied/modified from appsembler_api.utils in enterprise Ginkgo
+    Copied the template files from enterprise Ginkgo LMS templates
+    """
+    form = PasswordResetFormNoActive(request.data)
+    if form.is_valid():
+        form.save(
+            use_https=request.is_secure(),
+            from_email=configuration_helpers.get_value(
+                'email_from_address', settings.DEFAULT_FROM_EMAIL),
+            request=request,
+            domain_override=request.get_host(),
+            subject_template_name='appsembler/emails/set_password_subject.txt',
+            email_template_name='appsembler/emails/set_password_email.html')
+        return True
+    else:
+        return False
+
+
+class RegistrationViewSet(TahoeAuthMixin, viewsets.ViewSet):
```

Comment:
@estherjsuh I think we could inherit directly from v1 RegistrationViewSet class here, in that case you only need to rewrite the `create` method and there is no need to copy the `_normalize_bool_param` for example.

```suggestion
class RegistrationViewSet(v1.views.RegistrationViewSet):
```

Reply:
@melvinsoft I imported the `RegistrationViewSet` from v1, and updated v2 to:

- `class RegistrationViewSet(RegistrationViewSetV1):`